homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -75,51 +82,97 @@ async def async_setup_entry(
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.disarm
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                raise HomeAssistantError(
+                    f"Could not verify disarmed for {self._attr_name}: {error}"
+                ) from error
+
+        LOGGER.debug("Alarm disarmed: %s", alarm_state)
+        if alarm_state:
+            self._attr_state = STATE_MAP.get(YALE_STATE_DISARM)
+            self.async_write_ha_state()
+            return
+        raise HomeAssistantError("Could not disarm, check system ready for disarming.")
+
+    async def async_alarm_arm_home(self, code=None) -> None:
         """Send arm home command."""
-        self.coordinator.yale.arm_partial()
-
-    def alarm_arm_away(self, code=None):
+        if self._coordinator.yale:
+            try:
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.arm_partial
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                raise HomeAssistantError(
+                    f"Could not verify armed home for {self._attr_name}: {error}"
+                ) from error
+
+        LOGGER.debug("Alarm armed home: %s", alarm_state)
+        if alarm_state:
+            self._attr_state = STATE_MAP.get(YALE_STATE_ARM_PARTIAL)
+            self.async_write_ha_state()
+            return
+        raise HomeAssistantError("Could not arm home, check system ready for arming.")
+
+    async def async_alarm_arm_away(self, code=None) -> None:
         """Send arm away command."""
-        self.coordinator.yale.arm_full()
+        if self._coordinator.yale:
+            try:
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.arm_full
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                raise HomeAssistantError(
+                    f"Could not verify armed away for {self._attr_name}: {error}"
+                ) from error
+
+        LOGGER.debug("Alarm armed away: %s", alarm_state)
+        if alarm_state:
+            self._attr_state = STATE_MAP.get(YALE_STATE_ARM_FULL)
+            self.async_write_ha_state()
+            return
+        raise HomeAssistantError("Could not arm away, check system ready for arming.")
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_state = STATE_MAP.get(self.coordinator.data["alarm"])
```

Comment:
For coordinator entities it's normally better to use state properties than the shorthand attributes since it's usually just a matter of accessing the coordinator data and potentially do some formatting. Then we don't need to overwrite `_handle_coordinator_update`.